### PR TITLE
Upgrades to improved MapLibre Compose

### PR DESCRIPTION
**Pull request by Archdoog**
- Adds camera padding feature to correctly place user puck on landscape and portrait navigation view.
- Hides compass
- Moves attribution button and maplibre logo.
- Pitch works!
- Upgrades any maplibre types to `org.maplibre.android`

### CODE REVIEW

File: android/maplibreui/src/main/java/com/stadiamaps/ferrostar/maplibreui/extensions/CameraPadding.kt [hunk 1]
```diff
@@ -0,0 +1,29 @@
+package com.stadiamaps.ferrostar.maplibreui.extensions
+
+import com.maplibre.compose.camera.models.CameraPadding
+
+/**
+ * A camera padding configuration for a navigation map view that places the user puck in the lower
+ * center of the screen.
+ */
+fun CameraPadding.Companion.NavigationPortrait(): CameraPadding {
```
Comment (ianthetechie) [inline android/maplibreui/src/main/java/com/stadiamaps/ferrostar/maplibreui/extensions/CameraPadding.kt:9]: General Android lint I noticed... I know what you're going for here though ;) Not sure if it's the right way or not TBH. I won't block the PR for this but something to think about.

image

**Review by ianthetechie — COMMENTED** [review]
When testing this, I ran into some issues. An API 35 emulator crashed instantly (ok it's dev; whatever...). An API 34 emulator is also segfaulting. It's crashing in a GL thread, which suggests it's related to MapLibre (not 100% sure but seems likely here).

```
2024-07-16 16:27:14.329  9192-9267  libc                    com.stadiamaps.ferrostar.demo        A  Fatal signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 0x30 in tid 9267 (GLThread 71), pid 9192 (.ferrostar.demo)
```

At first I thought it was related to the padding but I'm not sure now.

Is this working for you? If so, can you give more config details? Here's my AVD config:

image

**Comment by Archdoog** [comment]
> When testing this, I ran into some issues. An API 35 emulator crashed instantly (ok it's dev; whatever...). An API 34 emulator is also segfaulting. It's crashing in a GL thread, which suggests it's related to MapLibre (not 100% sure but seems likely here).
> 
> ```
> 2024-07-16 16:27:14.329  9192-9267  libc                    com.stadiamaps.ferrostar.demo        A  Fatal signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 0x30 in tid 9267 (GLThread 71), pid 9192 (.ferrostar.demo)
> ```
> 
> At first I thought it was related to the padding but I'm not sure now.
> 
> Is this working for you? If so, can you give more config details? Here's my AVD config:
> 
> image

Looks like it's the pitch. Will try some things out to see if it's just a misuse bug on behalf of the compose playground.

File: android/maplibreui/src/main/java/com/stadiamaps/ferrostar/maplibreui/extensions/CameraPadding.kt (hunk 1 above)
Comment (Archdoog) [reply android/maplibreui/src/main/java/com/stadiamaps/ferrostar/maplibreui/extensions/CameraPadding.kt:9]: Yep. I've been ignoring those to keep it "kotlin enum like". I could put an actual suppression on it to note it's an intentional decision.

File: android/maplibreui/src/main/java/com/stadiamaps/ferrostar/maplibreui/extensions/CameraPadding.kt [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+  return CameraPadding(
+      start = 0.0,
+      top = 1300.0,
```
[the remark this answers is not in the document]
Comment (Archdoog) [reply android/maplibreui/src/main/java/com/stadiamaps/ferrostar/maplibreui/extensions/CameraPadding.kt]: This is gone and replaced by a much improved dynamic @Composable.

**Comment by Archdoog** [comment]
> When testing this, I ran into some issues. An API 35 emulator crashed instantly (ok it's dev; whatever...). An API 34 emulator is also segfaulting. It's crashing in a GL thread, which suggests it's related to MapLibre (not 100% sure but seems likely here).
> 
> ```
> 2024-07-16 16:27:14.329  9192-9267  libc                    com.stadiamaps.ferrostar.demo        A  Fatal signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 0x30 in tid 9267 (GLThread 71), pid 9192 (.ferrostar.demo)
> ```
> 
> At first I thought it was related to the padding but I'm not sure now.
> 
> Is this working for you? If so, can you give more config details? Here's my AVD config:
> 
> image

This branch has maplibre-compose 0.0.13, which reverts to maplibre 10. Testing proves the crash is not there.

**Review by ianthetechie — APPROVED** [review]

**Pull request merged**